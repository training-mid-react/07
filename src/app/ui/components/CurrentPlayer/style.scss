.current-player {
    font-size: 28px;
    padding: 10px;
    text-align: center;
    color: white;
    margin-top: 8px;
  }
  
  .current-player__player1 {
    color: white;
    font-weight: bold;
    background: linear-gradient(135deg, rgba(108, 108, 224, 0.589), rgba(168, 168, 207, 0.589));
    padding: 12px;
    border-radius: 4px;
  }
  
  .current-player__player2 {
    color: rgb(4, 94, 4);
    font-weight: bold;
    background: linear-gradient(135deg, rgb(123, 204, 123), rgb(4, 94, 4));
    padding: 12px;
    border-radius: 4px;
  }
  