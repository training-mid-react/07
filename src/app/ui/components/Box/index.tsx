import { CELL } from "@/app/core/interfaces/game.interface";
import "./style.scss"

interface Props {
    value: CELL;
    onClick: () => void;
}

const Box = ({ value, onClick }: Props) => {
    return (
        <div data-testid="box" className="box" onClick={onClick}>
            {
                value && <div style={{backgroundColor: value.color}} className="box__chip"></div>
            }
        </div>
    );
}

export default Box;