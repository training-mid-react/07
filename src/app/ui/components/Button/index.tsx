import React from 'react';
import './style.scss';

interface ButtonProps {
  children: React.ReactNode;
  className?: string;
  onClick: () => void;
  disabled?: boolean;
}

const Button: React.FC<ButtonProps> = ({ children, onClick, disabled, className }) => {
  return (
    <button data-testid='button-component' className={className || 'custom-button'} onClick={onClick} disabled={disabled}>
      {children}
    </button>
  );
};

export default Button;
