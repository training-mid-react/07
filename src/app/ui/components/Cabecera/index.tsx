
import "./style.scss";

interface Props {
    currentPlayer: number
}

const Cabecera = ({ currentPlayer }: Props) => {
    return (
        <>
            <section className="cabecera">
                <div>
                    <h2>Jugador 1</h2>
                    <div className="ficha-cabecera ficha-cabecera--jugador1"></div>
                </div>
                <div>
                    <h2>Jugador 2</h2>
                    <div className="ficha-cabecera ficha-cabecera--jugador2"></div>
                </div>
            </section>
            <h3>Turno del jugador: {currentPlayer}</h3>
        </>

    )
}

export default Cabecera 