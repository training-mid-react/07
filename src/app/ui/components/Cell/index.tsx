import { FC } from 'react';
import { PLAYER } from '../../../../core/interfaces';
import './style.scss';

export interface CellProps {
  value: PLAYER | null;
}

export const Cell: FC<CellProps> = ({ value }) => {

  return (
    <>
    <div className={`cell cell__${value}`} />
    <div className='cell__frame'></div>
    </>
  );
};
