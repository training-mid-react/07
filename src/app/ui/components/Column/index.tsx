import './style.scss';
import { useDropCell } from '../../../core/hooks/useDropCell';

interface ColumnProps {
  columnIndex: number;
  dropPiece: (column: number) => void;
  columnData: number[];
}

function Column({ columnIndex, dropPiece, columnData }: ColumnProps) {
  const { handleColumnClick, droppingCell } = useDropCell({ columnIndex, dropPiece, columnData })
  return (
    <div className="column" onClick={handleColumnClick}>
      {columnData.map((cell, rowIndex) => {
        const isDropping = droppingCell === rowIndex;
        const cellClass = `cell ${cell === 1 ? 'cell--player1' : cell === 2 ? 'cell--player2' : ''} ${isDropping ? 'cell--dropping' : ''}`;
        return (
          <div
            key={rowIndex}
            className={cellClass}
          />
        );
      })}
    </div>
  );
};

export default Column;
