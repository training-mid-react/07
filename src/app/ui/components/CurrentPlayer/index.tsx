import './style.scss';

interface ICurrentPlayerProps {
  currentPlayer: number;
}

export const CurrentPlayer = ({ currentPlayer }: ICurrentPlayerProps) => {
  return (
    <div className="current-player">
      Es el turno del Jugador: 
      <span className={currentPlayer === 1 ? 'current-player__player1' : 'current-player__player2'}>
        {currentPlayer}
      </span>
    </div>
  );
};
