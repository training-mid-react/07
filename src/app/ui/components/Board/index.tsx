import { FC } from 'react';
import { Column } from '../Column';
import { IBoard } from '../../../../core/interfaces';
import './style.scss';

interface Props {
  board: IBoard;
  handleDrop: (colIndex: number) => void;
}

export const Board:FC<Props> = ({board, handleDrop}) => {

  return (
    <div className="board">
      {board.map((_, colIndex) => (
        <Column key={colIndex} board={board} columnIndex={colIndex} handleDrop={handleDrop} />
      ))}
      <Column key={board.length} board={board} columnIndex={board.length} handleDrop={handleDrop} />
    </div>
  );
};
