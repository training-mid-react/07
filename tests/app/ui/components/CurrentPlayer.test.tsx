import { fireEvent, render } from "@testing-library/react";
import React from "react";
import { describe } from "node:test";
import { expect, test } from "vitest";
import { CurrentPlayer } from '../../../../src/app/ui/components/CurrentPlayer';

describe('Pruebas en el componente CurrentPlayer', () => {
    test('Match snapshot', () => {
        const mockPlayer = 1;
        render(<CurrentPlayer currentPlayer={mockPlayer} />)
        expect(document.body).toMatchSnapshot();
    })
    test('Match snapshot', () => {
        const mockPlayer = 2;
        render(<CurrentPlayer currentPlayer={mockPlayer} />)
        expect(document.body).toMatchSnapshot();
    })

    test('Revisa el texto del componente', () => {
        const { getByText } = render(<CurrentPlayer currentPlayer={1} />)
        expect(getByText('Es el turno del Jugador:')).toBeInTheDocument();
    })
    test('Revisa la clase del componente', () => {
        const { getByText } = render(<CurrentPlayer currentPlayer={1} />)
        expect(getByText('Es el turno del Jugador:')).toHaveClass('current-player');
    })
    test('debe mostrar el turno del Jugador 1 con la clase correcta', () => {
        const { getByText } = render(<CurrentPlayer currentPlayer={1} />);
        expect(getByText('Es el turno del Jugador:')).toBeInTheDocument();
        const playerSpan = getByText('1');
        expect(playerSpan).toBeInTheDocument();
        expect(playerSpan).toHaveClass('current-player__player1');
    });

    test('debe mostrar el turno del Jugador 2 con la clase correcta', () => {
        const { getByText } = render(<CurrentPlayer currentPlayer={2} />);
        expect(getByText('Es el turno del Jugador:')).toBeInTheDocument();
        const playerSpan = getByText('2');
        expect(playerSpan).toBeInTheDocument();
        expect(playerSpan).toHaveClass('current-player__player2');
    });
});