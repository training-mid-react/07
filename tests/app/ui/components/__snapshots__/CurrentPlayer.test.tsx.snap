// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Match snapshot 1`] = `
<body>
  <div>
    <div
      class="current-player"
    >
      Es el turno del Jugador:
      <span
        class="current-player__player1"
      >
        1
      </span>
    </div>
  </div>
</body>
`;

exports[`Match snapshot 2`] = `
<body>
  <div>
    <div
      class="current-player"
    >
      Es el turno del Jugador:
      <span
        class="current-player__player2"
      >
        2
      </span>
    </div>
  </div>
</body>
`;
